pandas/core/common.py

```diff
@@ -2510,6 +2510,8 @@ def is_floating_dtype(arr_or_dtype):
 
 
 def is_bool_dtype(arr_or_dtype):
+    if arr_or_dtype is NotImplemented:
```

Comment:
K, this seems more reasonable, though from my perspective I would probably still be happier explicitly handling `NotImplemented` inside the wrapper functions.